Remember last project per service/url

Phenome opened the pull request:
## What does this PR do?
This adds a new option to the settings page:
Remember last project per service/url

## How to test it
Try the option out. It should:
* Remember the last project used for said service (eg. github) or url
* If it's the same as default project from settings, use that one
* When changing the option, should wipe the mapping


Closes #210

syko commented on src/scripts/background.js:
```diff
@@ -286,7 +286,13 @@ var TogglButton = {
+      defaultProject = Db.getDefaultProject(),
+      rememberProjectPer = Db.get("rememberProjectPer");
+    if (rememberProjectPer) {
+      Db.setDefaultProject(entry.pid === defaultProject ? null : entry.pid,
```
This bit could use a comment. I went back and forth 6 times trying to figure out how this line works and doesn't mess up the global default project.

syko commented on src/html/settings.html:
```diff
@@ -61,6 +61,15 @@
               <div class="description">The default project will be selected when timer is started with no matched project</div>
             </li>
             <li>
+              <label for="remember-project-per">Remember project per: </label>
```
Minor thing but lose the `:`

Phenome commented:
Comments adressed:
* removed `:`
* Added comments
* Now tracking only when any default project is set. (yeah, I thought it was a bit odd to be tracking `pids`, but didn't want to change it.)
